pkg/model/crd.go

```diff
@@ -419,6 +438,85 @@ func (r *CRD) GoCodeSetInput(
 	return out
 }
 
+// GoCodeGetAttributesSetInput returns the Go code that sets the Input shape for a
+// resource's GetAttributes operation.
+//
+// As an example, for the GetTopicAttributes SNS API call, the returned code
+// looks like this:
+//
+// res.SetTopicArn(*r.ko.Status.ACKResourceMetadata.ARN)
+//
+// For the SQS API's GetQueueAttributes call, the returned code looks like this:
+//
+// res.SetQueueUrl(*r.ko.Status.QueueURL)
+//
+// You will note the difference due to the special handling of the ARN fields.
+func (r *CRD) GoCodeGetAttributesSetInput(
+	// String representing the name of the variable that we will grab the
+	// Input shape from. This will likely be "r.ko.Spec" since in the templates
+	// that call this method, the "source variable" is the CRD struct's Spec
+	// field which is used to populate the target variable, which is the Input
+	// shape
+	sourceVarName string,
+	// String representing the name of the variable that we will be **setting**
+	// with values we get from the Output shape. This will likely be
+	// "res" since that is the name of the "target variable" that the
+	// templates that call this method use for the Input shape.
+	targetVarName string,
+	// Number of levels of indentation to use
+	indentLevel int,
+) string {
+	op := r.Ops.GetAttributes
+	if op == nil {
+		return ""
+	}
+	inputShape := op.InputRef.Shape
+	if inputShape == nil {
+		return ""
+	}
+
+	out := "\n"
+	indent := strings.Repeat("\t", indentLevel)
+
+	for _, memberName := range inputShape.MemberNames() {
+		if r.IsPrimaryARNField(memberName) {
+			out += fmt.Sprintf(
+				"%s%s.Set%s(*%s.Status.ACKResourceMetadata.ARN)\n",
+				indent, targetVarName, memberName, sourceVarName,
+			)
+			continue
+		}
+
+		cleanMemberNames := names.New(memberName)
+		cleanMemberName := cleanMemberNames.Camel
+
+		sourceVarPath := sourceVarName
+		field, found := r.SpecFields[memberName]
+		if found {
+			sourceVarName = sourceVarPath + ".Spec." + cleanMemberName
```

Comment:
*sourceVarPath= sourceVarPath + ...

Reply:
nice catch. actually should be `sourceVarPath = sourceVaName + ".Spec." + cleanMemberName`